satellite/marketing/marketingweb/server_test.go

```diff
@@ -0,0 +1,86 @@
+package marketingweb
+
+import (
+	"net/http"
+	"testing"
+	"net/url"
+	"strings"
+	"log"
+)
+
+
+func buildForm(isReferralOffer bool) url.Values{
+	form := url.Values{}
+	form.Add("Name","May Credit")
+	form.Add("Description","desc")
+	form.Add("ExpiresAt","2019-06-27")
+	form.Add("InviteeCreditInCents","50")
+	form.Add("InviteeCreditDurationDays","50")
+	form.Add("RedeemableCap","150")
+
+	if isReferralOffer {
+		form.Add("AwardCreditInCents","50")
+		form.Add("AwardCreditDurationDays","50")
+	}
+	return form
+}
+
+func buildResources(endpoint string, isReferralOffer bool) (url.URL, url.Values){
+	URL, err := url.ParseRequestURI("http://127.0.0.1:10003")
+	if err != nil{
+		log.Fatalf("URL parsing Err : %v\n", err)
+	}
+	URL.Path = endpoint
+	form := buildForm(isReferralOffer)
+	URL.RawQuery = form.Encode()
+	return *URL, form
+}
+
+func callServer(t *testing.T,endpoint,params string, form url.Values) string{
+	c := http.Client{}
+	req, err := http.NewRequest("POST", endpoint, strings.NewReader(params))
+	if err != nil {
+		t.Fatalf("Err building request : %v\n", err)
```

Comment:
We probably wan to use `require.NoError` here

Reply:
pushed